src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -3,34 +3,41 @@
 import io.swagger.v3.oas.annotations.Operation;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 import ru.skypro.homework.dto.AdDto;
 import ru.skypro.homework.dto.AdsDto;
 import ru.skypro.homework.dto.CreateOrUpdateAdDto;
 import ru.skypro.homework.dto.ExtendedAdDto;
+import ru.skypro.homework.service.impl.AdServiceImpl;
 
+import javax.validation.Valid;
 import java.io.IOException;
-import java.util.ArrayList;
 
 @CrossOrigin(value = "http://localhost:3000")
 @RestController
 @RequestMapping("/ads")
 @Tag(name = "Объявления")
 @Validated
+@RequiredArgsConstructor
+
 public class AdsController {
+    private final AdServiceImpl adService;
```

Comment:
```suggestion
    private final AdService adService;
```
инъекцию делаем по интерфейсу всегда